overlord/hookstate/hookmgr.go

```diff
@@ -143,13 +153,38 @@ func (m *HookManager) doRunHook(task *state.Task, tomb *tomb.Tomb) error {
 		return err
 	}
 
-	// TODO: Actually dispatch the hook.
+	_, err = RunHook(setup.Snap, setup.Revision, setup.Hook)
+	if err != nil {
+		exitErr, ok := err.(*exec.ExitError)
+		if ok {
+			// "exit code 1" is not particularly helpful. So rewrite the error
+			// using stderr instead.
+			err = fmt.Errorf("%s", strings.Trim(string(exitErr.Stderr), " \n"))
```

Comment:
exitErr.Stderr is not defined and exec.Command().Ouptut() will only give you stdout, stderr goes to /dev/null AFAIK, see above for alternatives.